### PR TITLE
CIRC-197: Determine patron notice policy based on loan rules

### PR DESCRIPTION
**Pull request by jsavell**

### CODE REVIEW

**Comment by jsavell** [comment]
Looks like a PR got merged into master, creating conflicts for this one. 

I'll work on resolving the new conflicts.

File: ramls/circulation.raml [hunk 1]
```diff
@@ -406,6 +406,90 @@ resourceTypes:
             body:
               text/plain:
                 example: "Internal server error"
+    /notice-policy:
```
Comment (marcjohnson-kint) [inline ramls/circulation.raml]: Please update the `circulation` interface version (to 7.4) both here and in the module descriptor to reflect this addition.

File: src/test/java/api/CirculationRulesEngineAPITests.java [hunk 2]
```diff
@@ -67,7 +67,7 @@ private Policy applyRequestPolicy(ItemType itemType, String requestType,
       URL url = circulationRulesUrl(
           "/request-policy"
           + "?item_type_id="         + itemType.id
-          + "&request_type_id="         + requestType
+          + "&loan_type_id="         + requestType
```
Comment (marcjohnson-kint) [inline src/test/java/api/CirculationRulesEngineAPITests.java]: Why have the parameters for a determining a request policy changed?

File: descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -739,6 +749,16 @@
       "displayName": "Circulation - use circulation rules to get all matching request policies",
       "description": "Apply circulation rules to get all matching request policies"
     },
+    {
+      "permissionName": "circulation.rules.notice-policy.get",
```
Comment (marcjohnson-kint) [inline descriptors/ModuleDescriptor-template.json]: The endpoints (`/circulation/rules/patron-notice-policy` and `/circulation/rules/patron-notice-policy-all`?) relating to these policies don't seem to be described, please can they be added.

File: src/test/java/api/CirculationRulesEngineAPITests.java (hunk 2 above)
Comment (jsavell) [reply src/test/java/api/CirculationRulesEngineAPITests.java]: I'm not 100% sure, but the commit associated with the change is named 'Fixed broken tests', so it sounds like there was a bug.

https://github.com/folio-org/mod-circulation/commit/997506afc7aba5b7ecd7ed2bebef3bebf7e59d4a

File: descriptors/ModuleDescriptor-template.json (hunk 3 above)
Comment (jsavell) [reply descriptors/ModuleDescriptor-template.json]: Is this a different description than lines 754 - 761?

I'm not sure what's missing or where.

Comment (marcjohnson-kint) [reply descriptors/ModuleDescriptor-template.json]: Apologies, the endpoints for determining patron notice policies need defining for Okapi to use (and these permissions binding to them).

Similarly to https://github.com/folio-org/mod-circulation/blob/master/descriptors/ModuleDescriptor-template.json#L371-L394

Let me know if that doesn't make sense.

File: ramls/circulation.raml (hunk 1 above)
Comment (jsavell) [reply ramls/circulation.raml]: Done.

File: descriptors/ModuleDescriptor-template.json (hunk 3 above)
Comment (jsavell) [reply descriptors/ModuleDescriptor-template.json]: Ok, thanks. I've updated the handlers with the descriptions of the notice-policy endpoints.

File: src/test/java/org/folio/circulation/rules/Text2DroolsTest.java [hunk 4]
```diff
@@ -16,6 +20,8 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.CaseInsensitiveHeaders;
 import io.vertx.core.json.JsonArray;
 
 public class Text2DroolsTest {
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/circulation/rules/Text2DroolsTest.java]: There doesn't appear to be any new tests for patron notice policies, as there were with request policies.

Is that intentional?

**Pull request merged**